3 repo-contract tests fail on main after the Cohere2-MoE / GPT-OSS-Puzzle vendor commits

## What

On pristine main (verified at `dfb2f4b3` and still true at `0ca64412`), three repo-contract tests fail locally:

```
FAILED tests/test_mlx_compat.py::test_every_mlx_lm_consumer_installs_shim
  models/cohere2_moe.py (mlx_lm import @ line 14, install @ None)
FAILED tests/test_model_profiles_ssot.py::test_every_alias_has_explicit_profile_fields
FAILED tests/test_model_sizes.py::test_every_text_alias_has_a_manifest_entry
  missing from model_sizes.json: mlx-community/North-Mini-Code-1.0-4bit, mlx-community/North-Mini-Code-1.0-bf16
```

Introduced by the vendor commits aac18fb0 (#1223 North-Mini-Code Cohere2-MoE) and 837d4b78 (#1224 GPT-OSS Puzzle):

- `vllm_mlx/models/cohere2_moe.py` does a module-load-time `import mlx_lm` without the `_mlx_compat.install()` shim the contract requires (#404 M5 regression risk).
- The two new North-Mini-Code aliases are missing from `model_sizes.json` (the test's own remedy: `python3.12 scripts/gen_model_sizes.py`) and lack explicit profile fields.

## Why it matters

Every PR's local `full_unit` now carries these 3 pre-existing failures, so real regressions are harder to spot in the noise (three of my PR validation runs tonight had to explain them away). The CI `tests` gate apparently doesn't exercise these paths, which is why the commits landed green.

## Suggested fix

Mechanical, per each test's own remedy: add the shim install to `cohere2_moe.py` before its mlx_lm import (same pattern as `utils/mamba_cache.py`), run `scripts/gen_model_sizes.py`, and fill the explicit profile fields for the two new aliases. Left to the vendor-commit authors since it touches vendored model files and the alias SSOT.

--- a/tests/test_cohere2_moe_vendored.py
+++ b/tests/test_cohere2_moe_vendored.py
@@ -102,6 +102,7 @@ def test_checkpoint_profile_defaults_and_conservative_capabilities() -> None:
     assert profile is not None
     assert profile.tool_call_parser is None
     assert profile.reasoning_parser is None
+    assert profile.is_hybrid is False
     assert profile.is_moe is True
     assert profile.supports_spec_decode is False
 
@@ -118,5 +119,6 @@ def test_public_4bit_alias_defaults_and_conservative_capabilities() -> None:
     assert profile.hf_path == "mlx-community/North-Mini-Code-1.0-4bit"
     assert profile.tool_call_parser is None
     assert profile.reasoning_parser is None
+    assert profile.is_hybrid is False
     assert profile.is_moe is True
     assert profile.supports_spec_decode is False

--- a/vllm_mlx/aliases.json
+++ b/vllm_mlx/aliases.json
@@ -1,11 +1,13 @@
 {
   "north-mini-code-4bit": {
     "hf_path": "mlx-community/North-Mini-Code-1.0-4bit",
+    "is_hybrid": false,
     "is_moe": true,
     "supports_spec_decode": false
   },
   "north-mini-code-bf16": {
     "hf_path": "mlx-community/North-Mini-Code-1.0-bf16",
+    "is_hybrid": false,
     "is_moe": true,
     "supports_spec_decode": false
   },

--- a/vllm_mlx/model_sizes.json
+++ b/vllm_mlx/model_sizes.json
@@ -58,6 +58,8 @@
     "mlx-community/Muse-Glimmer-30B-bf16": 59581776079,
     "mlx-community/NVIDIA-Nemotron-3.5-Lightning-30B-A3B-4bit": 17792571825,
     "mlx-community/Nanbeige4.1-3B-4bit": 2231499570,
+    "mlx-community/North-Mini-Code-1.0-4bit": 18514853287,
+    "mlx-community/North-Mini-Code-1.0-bf16": 60988268172,
     "mlx-community/Phi-3.5-mini-instruct-4bit": 2152081350,
     "mlx-community/Qwen2.5-14B-Instruct-4bit": 8321091241,
     "mlx-community/Qwen3-0.6B-4bit": 351383618,

--- a/vllm_mlx/models/cohere2_moe.py
+++ b/vllm_mlx/models/cohere2_moe.py
@@ -11,6 +11,13 @@ from typing import Any
 
 import mlx.core as mx
 import mlx.nn as nn
+
+# Installing this before any ``mlx_lm`` import protects M5 single-stream
+# devices from mlx-lm's module-load-time thread-local stream capture (#404).
+from .. import _mlx_compat
+
+_mlx_compat.install()
+
 from mlx_lm.models.activations import swiglu
 from mlx_lm.models.base import (
     BaseModelArgs,

